src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioHostContextService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [ExportWorkspaceService(typeof(IHostContextService), ServiceLayer.Host), Shared]
+    internal class VisualStudioHostContextService : IHostContextService
+    {
+        public bool SolutionExistsAndFullyLoaded
+            => KnownUIContexts.SolutionExistsAndFullyLoadedContext.IsActive;
```

Comment:
> Is this ever true for Lightweight Solution load?

I would really hope not.  That would seem utterly and completely broken.  The point of UIContexts is so that features an accurately tell what state they're in, and wait until the host is actually in the right state for them to be doing work.  If LSL doesn't abide by this contract then that affects a lot of code that uses UIContext already in the LS.